app/_src/using-mesh/managing-ingress-traffic/builtin-listeners.md

```diff
@@ -0,0 +1,360 @@
+the port/protocol with other routes attached to other hostnames.
+
+{% tabs usage useUrlFragment=false %}
+{% tab usage Kubernetes %}
+
+```yaml
+spec:
+  ...
+  conf:
+    listeners:
+      - port: 8080
+        protocol: HTTP
+        hostname: foo.example.com
+        tags:
+          port: http/8080
+```
+
+{% endtab %}
+{% tab usage Universal %}
+
+```yaml
+conf:
+  listeners:
+    - port: 8080
+      protocol: HTTP
+      hostname: foo.example.com
+      tags:
+        port: http/8080
+```
+
+{% endtab %}
+{% endtabs %}
+
+In this example, the gateway proxy listens for HTTP protocol connections on TCP port 8080 but restricts the `Host` header to `foo.example.com`.
+
+{% tabs selectors useUrlFragment=false %}
+{% tab selectors Kubernetes %}
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshGateway
+mesh: default
+metadata:
+  name: edge-gateway
+spec:
+  selectors:
+    - match:
+        kuma.io/service: edge-gateway
+  conf:
+    listeners:
+      - port: 8080
+        protocol: HTTP
+        hostname: foo.example.com
+        tags:
+          vhost: foo.example.com
+      - port: 8080
+        protocol: HTTP
+        hostname: bar.example.com
+        tags:
+          vhost: bar.example.com
+```
+
+{% endtab %}
+{% tab selectors Universal %}
+
+```yaml
+type: MeshGateway
+mesh: default
+name: edge-gateway
+selectors:
+  - match:
+      kuma.io/service: edge-gateway
+conf:
+  listeners:
+    - port: 8080
+      protocol: HTTP
+      hostname: foo.example.com
+      tags:
+        vhost: foo.example.com
+    - port: 8080
+      protocol: HTTP
+      hostname: bar.example.com
+      tags:
+        vhost: bar.example.com
+```
+
+{% endtab %}
+{% endtabs %}
+
+Above shows a `MeshGateway` resource with two HTTP listeners on the same port.
+In this example, the gateway proxy will be configured to listen on port 8080, and accept HTTP requests for both hostnames.
+
+Note that because each listener entry has its own {{site.mesh_product_name}} tags, policy can still be targeted to a specific listener.
+{{site.mesh_product_name}} generates a set of tags for each listener by combining the tags from the listener, the `MeshGateway` and the `Dataplane`.
+{{ site.mesh_product_name}} matches policies against this set of combined tags.
+
+| `Dataplane` tags                 | Listener tags                              | Final Tags                                         |
+| -------------------------------- | ------------------------------------------ | -------------------------------------------------- |
+| kuma.io/service=edge-gateway     | vhost=foo.example.com                      | kuma.io/service=edge-gateway,vhost=foo.example.com |
+| kuma.io/service=edge-gateway     | kuma.io/service=example,domain=example.com | kuma.io/service=example,domain=example.com         |
+| kuma.io/service=edge,location=us | version=2                                  | kuma.io/service=edge,location=us,version=2         |
```

Comment:
nitpick: this does not render very nicely

![image](https://github.com/kumahq/kuma-website/assets/2753650/65c025e2-157c-46e9-aac6-afe35ac13fd5)

maybe we could split comma separated values to multiple lines? 🤷